sql/src/main/java/io/crate/execution/engine/collect/MapSideDataCollectOperation.java

```diff
@@ -49,36 +50,19 @@ public MapSideDataCollectOperation(CollectSourceResolver collectSourceResolver,
         this.threadPool = threadPool;
     }
 
-    /**
-     * dispatch by the following criteria:
-     * <p>
-     * * if local node id is contained in routing:<br>
-     * * if no shards are given:<br>
-     * &nbsp; -&gt; run row granularity level collect<br>
-     * &nbsp; except for doc level:
-     * &nbsp; &nbsp; if table if partitioned:
-     * &nbsp; &nbsp; -&gt; edge case for empty partitioned table
-     * &nbsp; &nbsp; else:
-     * &nbsp; &nbsp; -&gt; collect from information schema
-     * * if shards are given:<br>
-     * &nbsp; -&gt; run shard or doc level collect<br>
-     * * else if we got cluster RowGranularity:<br>
-     * &nbsp; -&gt; run node level collect (cluster level)<br>
-     * </p>
-     */
-    public CrateCollector createCollector(CollectPhase collectPhase,
-                                          RowConsumer consumer,
-                                          final CollectTask collectTask) {
+    public BatchIterator<Row> createIterator(CollectPhase collectPhase,
+                                             RowConsumer consumer,
```

Comment:
looks like only `requiresScroll` of the consumer is used here now, so we can just pass this?